### PR TITLE
Nvidia Flex application added

### PR DESCRIPTION
**Pull request by maceligueta**
It does not add any library or external file. Nvidia Flex must be installed first and provided to the configure.
Some cases needed a modification of the DEM application, which breaks the API of the Strategies.

### CODE REVIEW

**Comment by RiccardoRossi** [comment]
@maceligueta you are adding tetgen...

File: applications/NvidiaFlexApplication/license.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Kratos Multi-Physics - DEM Application
```
Comment (loumalouomega) [inline applications/NvidiaFlexApplication/license.txt]:
```suggestion
Kratos Multi-Physics - Nvidia FleX Application
```

**Comment by maceligueta** [comment]
Thanks @RiccardoRossi .  Let's put this accident in the list of problems brought by submodules...  (I did not add it explicitely! as far as I know).

**Comment by maceligueta** [comment]
Now I removed too much tetgen, I'm working on it...

**Comment by josep-m-carbonell** [comment]
I have reverted the submodule deletation.

**Review by salvalatorre — APPROVED** [review]
Looks OK...

**Pull request merged**